pkg/platform/aws/aws.go

```diff
@@ -146,6 +147,22 @@ func (c *config) Initialize(ex *terraform.Executor) error {
 		return err
 	}
 
+	// Extract control plane chart files to cluster assets directory.
```

Comment:
We had some ideas to completely remove `Initialize` function and replace it with `Render`, which would give the content of `cluster.tf` file.

I would suggest to keep things generic. The default charts are the same for all clusters, so they can be kept in a common place. Platforms might be able to expand this list, based on their configuration (add calico host protection, add/remove self-hosted kubelet). IMO platforms should not handle the extraction process itself, only define what needs to be extracted (as that must be platform-specific).

Reply:
>I would suggest to keep things generic. The default charts are the same for all clusters, so they can be kept in a common place.

See https://github.com/kinvolk/lokomotive/pull/655/commits/a508cb1c80e813b7bbd3f8d1daf7f704b9c60095#diff-f0398c2ff707cbe939f48da199c2bba9R27.

>IMO platforms should not handle the extraction process itself, only define what needs to be extracted (as that must be platform-specific).

In https://github.com/kinvolk/lokomotive/pull/655#discussion_r443656787 you've proposed that we extract the manifests in `Initialize()`, which is what I did... Anyway, I'm fine with re-evaluating this. Where would you suggest we extract control plane charts then?

Reply:
I would do the following:
>I'd do the unpacking in cli/cmd/cluster.go#initialize(), as we don't have a better place to do that right now.

And then use `Meta()` method to provide a list of extra charts to extract. This way we do extraction in exactly one place and we transfer only the information which is needed.

Reply:
I am not convinced about the `Meta()` abstraction - see https://github.com/kinvolk/lokomotive/pull/655#discussion_r444868433. Until we figure out what we want to do with the `Platform` interface I'm perfectly fine with duplicating the following lines among all platforms:

```go
	// Extract control plane chart files to cluster assets directory.
	for _, c := range platform.CommonControlPlaneCharts {
		src := filepath.Join(assets.ControlPlaneSource, c)
		dst := filepath.Join(assetDir, "cluster-assets", "charts", "kube-system", c)
		if err := assets.Extract(src, dst); err != nil {
			return errors.Wrapf(err, "Failed to extract charts")
		}
	}
```

I'd rather tolerate small-scale duplication than do ad-hoc modifications to our `Platform` abstraction without doing proper design discussions with the team.